### PR TITLE
Add Bulk Download Dialog

### PR DESCRIPTION
**Pull request by benomaire**
Possible use cases:
1. Download from Surah 18 to Surah 114
2. Download from Surah 78 to Surah 114
3. etc

This saves many touches for highlighting all these Surahs.

### CODE REVIEW

**Comment by benomaire** [comment]
Do you think using QuranSpinner (like audio_panel) is better?

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/BulkDownloadFragment.kt [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+  @Inject
+  lateinit var quranInfo: QuranInfo
+
+  private lateinit var suraFirstInput: ForceCompleteTextView
+  private lateinit var suraLastInput: ForceCompleteTextView
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    val activity: Activity? = activity
+    val inflater = activity!!.layoutInflater
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/BulkDownloadFragment.kt:43]:
```kotlin
val activity = requireActivity()
val inflater = activity.layoutInflater
```

File: app/src/main/java/com/quran/labs/androidquran/ui/fragment/BulkDownloadFragment.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        suras.contains(enteredFirstText) -> { enteredFirstText }
+        // leave to the next code
+        suraFirstAdapter.isEmpty -> { null }
+        // maybe first initialization or invalid input
+        else -> { suraFirstAdapter.getItem(0) }
+      }
+
+      var suraFirst = suras.indexOf(suraFirstName) + 1
+      // default to Al-Fatiha
+      if (suraFirst == 0) suraFirst = 1
```
Comment (ahmedre) [inline app/src/main/java/com/quran/labs/androidquran/ui/fragment/BulkDownloadFragment.kt:80]:
```kotlin
val suraFirst = max(suras.indexOf(suraFirstName) + 1, 1)
```

**Review by ahmedre — APPROVED** [review]
جزاكم الله خيراً

**Comment by ahmedre** [comment]
very minor comments but merging.

**Pull request merged**